src/solvers/dgsem_tree/subcell_limiters_2d.jl

```diff
@@ -13,21 +13,31 @@ function create_cache(limiter::Type{SubcellLimiterIDP}, equations::AbstractEquat
                                                                                    bound_keys)
 
     # Memory for bounds checking routine with `BoundsCheckCallback`.
-    # The first entry of each vector contains the maximum deviation since the last export.
-    # The second one contains the total maximum deviation.
-    idp_bounds_delta = Dict{Symbol, Vector{real(basis)}}()
+    # Local variable contains the maximum deviation since the last export.
+    # Using a threaded vector to parallelize bounds check.
+    idp_bounds_delta_local = Dict{Symbol, Vector{real(basis)}}()
+    # Global variable contains the total maximum deviation.
+    idp_bounds_delta_global = Dict{Symbol, real(basis)}()
+    # Note: False sharing causes critical performance issues on multiple threads when using a vector
+    # of length `Threads.nthreads()`. Initializing a vector of length `n * Threads.nthreads()`
+    # and then only using every n-th entry, fixes the problem and allows proper scaling.
+    # Since there are no processors with caches over 128B, we use `n = 128B / size(uEltype)`
+    stride_size = div(128, sizeof(eltype(basis.nodes))) # = n
```

Comment:
Thanks a lot for the informative comment! It would be great if you could add something like this to the devdocs of Trixi.jl for future reference (but in another PR)

Reply:
Should I still do this, although the conservation is resolved :sweat_smile: Do you refer to [this](https://trixi-framework.github.io/Trixi.jl/dev/development/) as devdocs? Isn't there only very informative and general information about the use of Trixi.jl?

Reply:
It would be great if you can find the time, yes, but not necessarily before this PR is merged. I was thinking of the performance docs: https://trixi-framework.github.io/Trixi.jl/dev/performance/